### PR TITLE
blueprints for creating GKE, bucket, folders, projects

### PR DESCRIPTION
**Pull request by hilliao**
gke/ - create a GKE cluster and a bucket
folder/ - create a folder in a given folder, then create a project in the created folder.

### CODE REVIEW

File: folder/Kptfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+openAPI:
+  definitions:
+    io.k8s.cli.setters.name:
+      x-k8s-cli:
+        setter:
+          name: name
+          value: ${FOLDER_NAME}
+          required: true
+      description: folder name
+    io.k8s.cli.substitutions.folder-name:
```
Comment (morgante) [inline folder/Kptfile:14]: You shouldn't need a substitution here—the setter can be referenced directly.

File: folder/Kptfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          - marker: ${NAME}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.setters.project-id-in-folder:
+      x-k8s-cli:
+        setter:
+          name: project-id-in-folder
+          value: ${PROJECT_ID}
+          required: true
+      description: project ID
+    io.k8s.cli.setters.folder-id:
```
Comment (morgante) [inline folder/Kptfile:29]: For clarity, I'd suggest calling this parent-folder-id.

Comment (hilliao) [reply folder/Kptfile]: changed

File: folder/Kptfile (hunk 1 above)
Comment (hilliao) [reply folder/Kptfile:14]: using a prefix in folder name

**Comment by hilliao** [comment]
examples not accepted because they are not generic enough

**Comment by karlkfi** [comment]
GKE: https://github.com/GoogleCloudPlatform/blueprints/pull/16
Project: https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/project
Folder Hierarchy: https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/hierarchy

**Pull request closed without merging**